### PR TITLE
Update regex that checks the os-release in the container

### PR DESCRIPTION
**Pull request by b10n1k**
Include optional Beta1 value in the pretty_name variable.

- [Verification run](http://aquarius.suse.cz/tests/4041#details)

### CODE REVIEW

**Review by Vogtinator — CHANGES_REQUESTED** [review]
This can be anything, from `(Snapshot42)` over `(Public Beta 1)` to being empty.

As `BETA=1` is set, can you add a match for `\([^)]+\)` (or so) in those cases?

That way it's ensured that the beta marker is only there iff it is a beta build.

**Review by pdostal — APPROVED** [review]

File: lib/containers/container_images.pm [hunk 1]
```diff
@@ -66,7 +66,7 @@ sub build_with_zypper_docker {
     # zypper docker can only update image if version is same as SUT
     if ($distri eq 'sle') {
         my $pretty_version = $version =~ s/-SP/ SP/r;
-        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version"/ });
+        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version(\s\([^)].+\))"/ });
```
Comment (Vogtinator) [inline lib/containers/container_images.pm:69]: The `.` is wrong there:

```suggestion
        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version(\s\([^)]+\))"/ });
```

**Review by Vogtinator — CHANGES_REQUESTED** [review]
This expects the beta string unconditionally now

File: lib/containers/container_images.pm (hunk 1 above)
Comment (b10n1k) [reply lib/containers/container_images.pm]: i am not sure if this makes much difference. but i have updated

File: lib/containers/container_images.pm [hunk 2]
```diff
@@ -66,7 +66,7 @@ sub build_with_zypper_docker {
     # zypper docker can only update image if version is same as SUT
     if ($distri eq 'sle') {
         my $pretty_version = $version =~ s/-SP/ SP/r;
-        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version"/ });
+        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version(\s\([^)]+\))?"/ });
```
Comment (Vogtinator) [inline lib/containers/container_images.pm:69]:
```suggestion
        $betaversion = get_var('BETA') ? '\s\([^)]+\)' : '';
        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server ${pretty_version}${betaversion}"/ });
```

Comment (b10n1k) [reply lib/containers/container_images.pm:69]: yea i was not sure if this was necessary when i created the regex. let me update it.

**Comment by Vogtinator** [comment]
A verification run for a product in beta and a product not in beta would be useful, just in case.

**Comment by jlausuch** [comment]
This PR breaks tests: https://openqa.suse.de/tests/5063516

File: lib/containers/container_images.pm [hunk 3]
```diff
@@ -66,7 +66,8 @@ sub build_with_zypper_docker {
     # zypper docker can only update image if version is same as SUT
     if ($distri eq 'sle') {
         my $pretty_version = $version =~ s/-SP/ SP/r;
-        validate_script_output("$runtime container run --entrypoint '/bin/bash' --rm $image -c 'cat /etc/os-release'", sub { /PRETTY_NAME="SUSE Linux Enterprise Server $pretty_version"/ });
+        $betaversion = get_var('BETA') ? '\s\([^)]+\)' : '';
```
Comment (jlausuch) [inline lib/containers/container_images.pm:69]:
```suggestion
        my $betaversion = get_var('BETA') ? '\s\([^)]+\)' : '';
```

File: lib/containers/container_images.pm [hunk 4]
```diff
@@ -105,9 +106,10 @@ sub test_opensuse_based_image {
     # It is the right version
     if ($distri eq 'sle') {
         my $pretty_version = $version =~ s/-SP/ SP/r;
+        $betaversion = get_var('BETA') ? '\s\([^)]+\)' : '';
```
Comment (jlausuch) [inline lib/containers/container_images.pm:109]:
```suggestion
        my $betaversion = get_var('BETA') ? '\s\([^)]+\)' : '';
```

**Comment by jlausuch** [comment]
Looks like it fixes the problem we had: 
https://openqa.suse.de/tests/5064243#
https://openqa.suse.de/tests/5064241#
The failure you see is due to another issue in another test. So I guess ok to merge this when Travis is done.

**Review by jlausuch — APPROVED** [review]

**Comment by Vogtinator** [comment]
Is there any verification run for a product not in beta?

**Comment by jlausuch** [comment]
> Is there any verification run for a product not in beta?

on the way https://openqa.suse.de/tests/5064862

**Pull request merged**